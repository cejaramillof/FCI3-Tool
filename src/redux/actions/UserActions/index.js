import * as TYPES from './UserActions.types';


export function signIn({ email, name, lastname }) {
  return {
    type: TYPES.SIGN_IN,
    user: {
      email, name, lastname
    }
  }
}
