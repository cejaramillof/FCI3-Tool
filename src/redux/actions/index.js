import { UPDATE_COMPETENCE, UPDATE_LIST, UPDATE_TAG_COLLECTION_COMPETENCE , NEXT_COMPETENCE, UPDATE_SUMMARY, UPDATE_USER, UPDATE_LANGUAGE } from './../types';

export function updateTagList({id, tagsSelected, level}) {
  return {
      type: UPDATE_LIST,
      competence: { 
        competenceId: id,
        tagsSelected,
        level
      }
    }
}

export function nextCompetence() {
  return {
      type: NEXT_COMPETENCE
    }
}

export function updateLanguage(lang) {
  return {
    type: UPDATE_LANGUAGE,
    lang
  }
}

export function updateCompetence({ id, description }) {
  return {
      type: UPDATE_COMPETENCE,
      competence:{
        id,
        description
      }
    }
}

export function updateTagCollectionCompetence({id, tags}) {
  return {
      type: UPDATE_TAG_COLLECTION_COMPETENCE,
      competence:{
        idSelectedCC: id,
        tagsCollection: tags
      }
    }
}

export function updateUser (user) {
  return {
      type: UPDATE_USER,
      userData:{
        user
      }
    }
}


export function updateSummary () {
  return {
    type: UPDATE_SUMMARY
  }
}
