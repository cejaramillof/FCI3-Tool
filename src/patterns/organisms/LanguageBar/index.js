import React, { useState } from 'react';
import LanguageButton from '../../molecules/LanguageButton';
import { connect } from 'react-redux';
import { updateLanguage } from '../../../redux/actions/index'
import i18n from 'i18n';

function LanguageBar(props) {
    const [lang, setLg] = useState(i18n.language);

    const changeLanguage =  (lang) => {
        props.updateLanguage(lang);
        i18n.changeLanguage(lang);
        setLg(lang);
    }

    return (
        <div className="col-sm-12 col-md-3 col-lg-3 d-none d-sm-block">
            <div className="btn-group language-selector" role="group" aria-label="Third group">
                <LanguageButton lang="en" changeLanguage={changeLanguage} />
                <LanguageButton lang="es" changeLanguage={changeLanguage} />
            </div>
        </div>
    )

}

const mapDispatchToProps = { updateLanguage };

export default connect(
    null, 
    mapDispatchToProps
)(LanguageBar);

