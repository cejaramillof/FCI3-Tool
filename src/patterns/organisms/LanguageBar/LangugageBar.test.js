import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import LanguageBar from './index';
import { updateLanguage } from '../../../redux/actions/index';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

describe('LanguageBar test english', () => {
    const props = {
        updateLanguage: jest.fn(),
        onClick: sinon.spy(),
        currentLang: 'en'
    }
    
    const payload = {
        type: 'UPDATE_LANGUAGE',
        lang: 'en'

    }

    const mockStore = configureStore();
    const store = mockStore();

    store.dispatch(updateLanguage()); 

    const wrapper = shallow(
        <Provider store={store}>
            <LanguageBar {...props} store={store} />
        </Provider>
    );


    
    it('Should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should despatch actions', () => expect(updateLanguage('en')).toEqual(payload))
});

describe('LanguageBar test espanish', () => {
    const props = {
        updateLanguage: jest.fn(),
        onClick: sinon.spy(),
        currentLang: 'es'
    }
    
    const payload = {
        type: 'UPDATE_LANGUAGE',
        lang: 'es'

    }

    const mockStore = configureStore();
    const store = mockStore();

    store.dispatch(updateLanguage()); 

    const wrapper = shallow(
        <Provider store={store}>
            <LanguageBar {...props} store={store} />
        </Provider>
    );


    
    it('Should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('Should despatch actions', () => expect(updateLanguage('es')).toEqual(payload))
});
