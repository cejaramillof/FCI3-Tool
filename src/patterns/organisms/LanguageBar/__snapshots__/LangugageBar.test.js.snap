// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageBar test english Should render correctly 1`] = `ShallowWrapper {}`;

exports[`LanguageBar test espanish Should render correctly 1`] = `ShallowWrapper {}`;
